File: FamilyTools.Web/src/app/models/account-enter.ts
import { BaseModel } from './base-model';
import { AccountLine } from './account-line';
import { AccountTag } from './account-tag';

export class AccountEnter extends BaseModel {
  lines: AccountLine[];
  tag: AccountTag;
  name: string;
  totalValue: number;
  date: Date;

  constructor(name: string, totalValue: number, tag: AccountTag, date: Date, lines: AccountLine[], id?: number, creationDate?: Date, updateDate?: Date){
    super(id, creationDate, updateDate);
    this.lines = lines;
    this.tag = tag;
    this.name = name;
    this.totalValue = totalValue;
    this.date = date;
  }

    getTotal() : number{
      return this.lines.reduce((acc, line) => acc + line.value, 0)
    }
}

File: FamilyTools.EasyCompta/Model/AccountEnter.cs
using System.ComponentModel.DataAnnotations.Schema;

using FamilyTools.EasyCompta.Model;

namespace FamilyTools.EasyCompta.Model
{
    [Table("AccountEnters")]
    public class AccountEnter : BaseModel
    {
        public List<AccountLine> Lines { get; set; }

        public AccountTag Tag { get; set; }

        public string Name { get; set; }

        public float TotalValue { get; set; }

        public DateOnly LifeTime { get; set; }

        public AccountEnter()
        {
            if (this.Lines?.Count > 0)
            {
                this.TotalValue = this.Lines.Sum(line => line.Value);
            }
        }
    }
}
